File: lib/l10n/sort_arb.py
import json

file_names = [
    "app_en.arb",
    "app_fr.arb",
]
for file_name in file_names:
    with open(f"lib/l10n/{file_name}", "r", encoding="utf-8") as f:
        content = json.load(f)
    sorted_content = dict(sorted(content.items()))
    with open(f"lib/l10n/{file_name}", "w", encoding="utf-8") as f:
        json.dump(sorted_content, f, ensure_ascii=False, indent=2)

File: lib/l10n/sort_arb.dart
import 'dart:convert';
import 'dart:io';

void main() async {
  final fileNames = ["app_en.arb", "app_fr.arb"];

  for (final fileName in fileNames) {
    final file = File("lib/l10n/$fileName");
    final content =
        jsonDecode(await file.readAsString()) as Map<String, dynamic>;

    final sortedKeys = content.keys.toList()
      ..sort((a, b) {
        final keyA =
            a.toLowerCase().replaceAll("@", "") +
            (a.startsWith("@") ? "a" : "");
        final keyB =
            b.toLowerCase().replaceAll("@", "") +
            (b.startsWith("@") ? "a" : "");
        return keyA.compareTo(keyB);
      });

    final sortedContent = {for (final key in sortedKeys) key: content[key]};

    await file.writeAsString(
      const JsonEncoder.withIndent('  ').convert(sortedContent),
    );
  }
}
